blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultibranchPipelineRunContainer.java

```diff
@@ -84,21 +84,24 @@ public BlueRun get(String name) {
             branches = Lists.newArrayList(blueMbPipeline.getBranches().list());
             sortBranchesByLatestRun(branches);
         }
+        Iterator<BlueRun> allRunIt = Collections.emptyIterator();
 
         for (final BluePipeline b : branches) {
             BlueRunContainer blueRunContainer = b.getRuns();
             if(blueRunContainer==null){
                 continue;
             }
             Iterator<BlueRun> it = blueRunContainer.iterator(0, MAX_MBP_RUNS_ROWS);
-            int count = 0;
-            Utils.skip(it, start);
-            while (it.hasNext() && count++ < limit) {
-                c.add(it.next());
-            }
+            allRunIt = Iterators.concat(allRunIt,it);
         }
 
+
         Collections.sort(c, LATEST_RUN_START_TIME_COMPARATOR);
+        Utils.skip(allRunIt, start);
+        int count = 0;
+        while (allRunIt.hasNext() && count++ < limit) {
```

Comment:
I think it would be better to keep the limit in the `branches` loop, without that, you're potentially reading a lot more stuff from disk (and slow/etc.)

Reply:
Looping in a branches can keep running fast, but in some cases some runs cannot be listed correctly.